api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java

```diff
@@ -97,17 +97,30 @@ private EligibilityAndEntitlementDecision buildDecision(EligibilityResponse elig
                                                             PaymentCycleVoucherEntitlement entitlement,
                                                             UUID existingClaimId) {
         EligibilityStatus eligibilityStatus = determineEligibilityStatus(eligibilityResponse, entitlement);
+        PaymentCycleVoucherEntitlement voucherEntitlement = determinePaymentCycleVoucherEntitlementFromStatus(entitlement, eligibilityStatus);
         return EligibilityAndEntitlementDecision.builder()
                 .eligibilityStatus(eligibilityStatus)
                 .qualifyingBenefitEligibilityStatus(QualifyingBenefitEligibilityStatus.fromEligibilityStatus(eligibilityResponse.getEligibilityStatus()))
-                .voucherEntitlement(entitlement)
```

Comment:
Would it not be better to do the check earlier and not even calculate the entitlement if the response is not ELIGIBLE?

Reply:
The only problem with this is that we'd need to check again later on when we decide that they are INELIGIBLE because they are entitled to no vouchers, so makes sense to do it at the end.